plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -431,8 +431,10 @@ public boolean getDeployToAllNodes(String modelId) {
     }
 
     public void setModelInfo(String modelId, MLModel mlModel) {
-        MLModelCache mlModelCache = getExistingModelCache(modelId);
-        mlModelCache.setModelInfo(mlModel);
+        MLModelCache mlModelCache = modelCaches.get(modelId);
+        if (mlModelCache != null) {
```

Comment:
@zane-neo , we should check if model cache exist or not

Reply:
I think we also need to fine tuning the error message in `getExistingModelCache` method, when model is not in cache, it should be a not deployed model, we should return the should deploy model info, so that in the future other method can use this method safely.

Reply:
For current code, I think either local or remote needs to be deployed first, are you saying some feature will be implemented in the future?